conftest.py

```diff
@@ -79,6 +79,31 @@ def mock_basket(settings):
         responses.POST,
         settings.BASKET_URL + '/news/unsubscribe/{}/'.format(USER_TOKEN),
         json={'status': 'ok', 'token': USER_TOKEN})
+    responses.add(
```

Comment:
I wasn't sure if I should've went this route either, the errors in the tests stated the response did not match any registered mocks that led me here. I agree that a new fixture should be created if this is the way to go though.

Reply:
Yeah, don't add them here - this is for urls that are pinged site-wide (if we're randomly accessing the API elsewhere on the site it's a bug and we certainly want those tests to error).  Add the to the TestXxx setup for tests that deal with shelves.

Reply:
If we're accessing the api in the test instance we don't need any of these responses adding now - we can just add the addons (using `addon_factory()`) in the test setup instead.

Reply:
Though for the rest of the tests that don't _specifically_ test validation it would be easier to just mock the whole validation, like @bobsilverberg suggested.  You can use `mock.patch` on the clean method (see other comments where I talk about using clean over the validator method)